src/arbitraries/types.ts

```diff
@@ -8,3 +8,8 @@ export type ArbitrarySize = {
   type: 'exact' | 'estimated'
   credibleInterval?: [number, number]
 }
+
+export type MappedArbitraryConfig<A,B> = {
+  inverseFunction?: (b: B) => A | A[],
+  customCanGenerate?: (pick: FluentPick<B>) => boolean
```

Comment:
I would simply change the name to `inverseMap` and `canGenerate`. I also don't like the name `MappedArbitraryConfig`, but can't think of a better alternative xD

Reply:
By the way, this solution you've conjured is very ingenious, as it supports passing inline overrides in those cases! The downside is that it becomes too verbose...

Reply:
Already renamed the variables as you suggested!